TestNewSSHClient: remote command exited without exit status or exit signal

https://github.com/kubernetes/minikube/pull/7282/checks?check_run_id=538693321

```
--- FAIL: TestNewSSHClient (0.25s)
    ssh_mock.go:176: Listening on 127.0.0.1:35513
    ssh_mock.go:83: Accepting...
    ssh_mock.go:83: Accepting...
    ssh_mock.go:135: exec request received: &{Type:exec WantReply:true Payload:[0 0 0 3 102 111 111] ch:0xc0001ec6c0 mux:<nil>}
    sshutil_test.go:67: Error running "foo": wait: remote command exited without exit status or exit signal
    ssh_mock.go:192: Stopping
panic: Fail in goroutine after TestNewSSHClient has completed

goroutine 5 [running]:
testing.(*common).Fail(0xc0001cf320)
	/usr/local/go1.14/src/testing/testing.go:613 +0x11a
testing.(*common).Errorf(0xc0001cf320, 0xc32961, 0x16, 0xc000187f00, 0x1, 0x1)
	/usr/local/go1.14/src/testing/testing.go:712 +0x90
k8s.io/minikube/pkg/minikube/tests.(*SSHServer).handleRequest(0xc0001217a0, 0xd49840, 0xc0001ec6c0, 0xc000159f80)
	/home/runner/work/minikube/minikube/pkg/minikube/tests/ssh_mock.go:154 +0x381
k8s.io/minikube/pkg/minikube/tests.(*SSHServer).handleIncomingConnection(0xc0001217a0, 0xd4c9c0, 0xc000010018)
	/home/runner/work/minikube/minikube/pkg/minikube/tests/ssh_mock.go:118 +0x15f
created by k8s.io/minikube/pkg/minikube/tests.(*SSHServer).serve
	/home/runner/work/minikube/minikube/pkg/minikube/tests/ssh_mock.go:92 +0x4a
FAIL	k8s.io/minikube/pkg/minikube/sshutil	0.261s
```

Travis:
`ok  	k8s.io/minikube/pkg/minikube/sshutil	0.525s	coverage: 100.0% of statements`

--- a/pkg/minikube/sshutil/sshutil_test.go
+++ b/pkg/minikube/sshutil/sshutil_test.go
@@ -36,6 +36,7 @@ func TestNewSSHClient(t *testing.T) {
 	if err != nil {
 		t.Fatalf("NewSSHServer: %v", err)
 	}
+
 	port, err := s.Start()
 	if err != nil {
 		t.Fatalf("Error starting ssh server: %v", err)
@@ -50,27 +51,30 @@ func TestNewSSHClient(t *testing.T) {
 		},
 		T: t,
 	}
+
 	c, err := NewSSHClient(d)
 	if err != nil {
-		t.Fatalf("Unexpected error: %v", err)
+		t.Fatalf("Error creating client: %v", err)
 	}
 	defer c.Close()
 
 	sess, err := c.NewSession()
 	if err != nil {
-		t.Fatal("Error creating new session for ssh client")
+		t.Fatalf("Error creating new session: %v", err)
 	}
 	defer sess.Close()
 
 	cmd := "foo"
 	if err := sess.Run(cmd); err != nil {
-		t.Fatalf("Error running %q: %v", cmd, err)
+		t.Errorf("Error running %q: %v", cmd, err)
 	}
+
 	if !s.Connected {
-		t.Fatalf("Server not connected")
+		t.Errorf("mock ssh server is not connected")
 	}
+
 	if _, ok := s.Commands[cmd]; !ok {
-		t.Fatalf("Expected command: %s", cmd)
+		t.Errorf("Expected %q to be run, but it never was!", cmd)
 	}
 }
 

--- a/pkg/minikube/tests/ssh_mock.go
+++ b/pkg/minikube/tests/ssh_mock.go
@@ -24,6 +24,7 @@ import (
 	"io"
 	"net"
 	"strconv"
+	"sync"
 	"sync/atomic"
 	"testing"
 
@@ -79,8 +80,11 @@ type execRequest struct {
 
 // Serve loop, listen for connections and store the commands.
 func (s *SSHServer) serve() {
+	s.t.Logf("Serving ...")
+	loop := 0
 	for {
-		s.t.Logf("Accepting...")
+		loop++
+		s.t.Logf("[loop %d] Accepting for %v...", loop, s)
 		c, err := s.listener.Accept()
 		if s.quit {
 			return
@@ -95,13 +99,19 @@ func (s *SSHServer) serve() {
 
 // handle an incoming ssh connection
 func (s *SSHServer) handleIncomingConnection(c net.Conn) {
+	var wg sync.WaitGroup
+
 	_, chans, reqs, err := ssh.NewServerConn(c, s.Config)
 	if err != nil {
 		s.t.Logf("newserverconn error: %v", err)
 		return
 	}
 	// The incoming Request channel must be serviced.
-	go ssh.DiscardRequests(reqs)
+	wg.Add(1)
+	go func() {
+		ssh.DiscardRequests(reqs)
+		wg.Done()
+	}()
 
 	// Service the incoming Channel channel.
 	for newChannel := range chans {
@@ -115,12 +125,15 @@ func (s *SSHServer) handleIncomingConnection(c net.Conn) {
 		}
 		s.Connected = true
 		for req := range requests {
-			s.handleRequest(channel, req)
+			s.handleRequest(channel, req, &wg)
 		}
 	}
+	wg.Wait()
 }
 
-func (s *SSHServer) handleRequest(channel ssh.Channel, req *ssh.Request) {
+func (s *SSHServer) handleRequest(channel ssh.Channel, req *ssh.Request, wg *sync.WaitGroup) {
+	wg.Add(1)
+
 	go func() {
 		// Explicitly copy buffer contents to avoid data race
 		b := s.Transfers.Bytes()
@@ -128,6 +141,7 @@ func (s *SSHServer) handleRequest(channel ssh.Channel, req *ssh.Request) {
 			s.t.Errorf("copy failed: %v", err)
 		}
 		channel.Close()
+		wg.Done()
 	}()
 
 	switch req.Type {
@@ -144,12 +158,15 @@ func (s *SSHServer) handleRequest(channel ssh.Channel, req *ssh.Request) {
 		}
 		s.Commands[cmd.Command] = 1
 
+		s.t.Logf("returning output for %s ...", cmd.Command)
 		// Write specified command output as mocked ssh output
 		if val, err := s.GetCommandToOutput(cmd.Command); err == nil {
 			if _, err := channel.Write([]byte(val)); err != nil {
 				s.t.Errorf("Write failed: %v", err)
 			}
 		}
+
+		s.t.Logf("setting exit-status for %s ...", cmd.Command)
 		if _, err := channel.SendRequest("exit-status", false, []byte{0, 0, 0, 0}); err != nil {
 			s.t.Errorf("SendRequest failed: %v", err)
 		}

